### PR TITLE
Document exceptions on async Socket methods.

### LINKED ISSUE #1350: Needs exceptions documented
What exceptions does this method throw? 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 659f9aff-32c9-589d-2e46-2742a3babaf8
* Version Independent ID: 2d725c88-6146-437d-6af6-7416089daeb0
* Content: [SocketTaskExtensions.ReceiveAsync Method (System.Net.Sockets)](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.sockettaskextensions.receiveasync?view=netstandard-2.0#System_Net_Sockets_SocketTaskExtensions_ReceiveAsync_System_Net_Sockets_Socket_System_ArraySegment_System_Byte__System_Net_Sockets_SocketFlags_)
* Content Source: [xml/System.Net.Sockets/SocketTaskExtensions.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.Net.Sockets/SocketTaskExtensions.xml)
* Product: **dotnet-api**
* GitHub Login: @karelz
* Microsoft Alias: **ncldev**

### PR DESCRIPTION
**Pull request by rzikm**
Fixes #1350.

### CODE REVIEW

File: xml/System.Net.Sockets/Socket.xml [hunk 1]
```diff
@@ -444,6 +444,13 @@ This method populates the <xref:System.Net.Sockets.Socket> instance with data ga
         <summary>Accepts an incoming connection.</summary>
         <returns>An asynchronous task that completes with the accepted Socket.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.InvalidOperationException">An invalid operation was requested. This exception occurs if the accepting <see cref="T:System.Net.Sockets.Socket" /> is not listening for connections or the accepted socket is bound.
+
+ You must call the <see cref="M:System.Net.Sockets.Socket.Bind(System.Net.EndPoint)" /> and <see cref="M:System.Net.Sockets.Socket.Listen(System.Int32)" /> method before calling the <see cref="M:System.Net.Sockets.Socket.AcceptAsync(System.Net.Sockets.SocketAsyncEventArgs)" /> method.
+
+ This exception also occurs if the socket is already connected or a socket operation was already in progress using the specified <paramref name="e" /> parameter.</exception>
+        <exception cref="T:System.Net.Sockets.SocketException">An error occurred when attempting to access the socket.</exception>
```
Comment (antonfirsov) [inline xml/System.Net.Sockets/Socket.xml:452]: I believe we throw this only asynchronously from most (if not all) the methods. Do we document async exceptions similarly in other Networking areas?

Comment (rzikm) [reply xml/System.Net.Sockets/Socket.xml:452]: I think so, e.g. HttpClient has anotations for HttpRequestException on async methods (https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.sendasync?view=net-7.0)

Comment (rzikm) [reply xml/System.Net.Sockets/Socket.xml:452]: same for AuthenticationException on SslStream.AuthenticateAs*Async (https://learn.microsoft.com/en-us/dotnet/api/system.net.security.sslstream.authenticateasclientasync?view=net-6.0)

Comment (antonfirsov) [reply xml/System.Net.Sockets/Socket.xml:452]: Looks like BCL is very inconsistent on this -- https://github.com/dotnet/dotnet-api-docs/issues/7840#issuecomment-1282761648, but at least you are making this consistent with the rest of the networking docs 😆

File: xml/System.Net.Sockets/Socket.xml [hunk 2]
```diff
@@ -4029,6 +4068,17 @@ This method populates the <xref:System.Net.Sockets.Socket> instance with data ga
...
+        <exception cref="T:System.ArgumentOutOfRangeException">
+          <paramref name="port" /> is less than <see cref="F:System.Net.IPEndPoint.MinPort" />.  
+  
+ -or-  
+  
+ <paramref name="port" /> is greater than <see cref="F:System.Net.IPEndPoint.MaxPort" />.</exception>
+        <exception cref="T:System.InvalidOperationException">The <see cref="T:System.Net.Sockets.Socket" /> is listening.</exception>
+        <exception cref="T:System.Net.Sockets.SocketException">An error occurred when attempting to access the socket.</exception>
+        <exception cref="T:System.ObjectDisposedException">The <see cref="T:System.Net.Sockets.Socket" /> has been closed.</exception>
+        <exception cref="T:System.Security.SecurityException">A caller higher in the call stack does not have permission for the requested operation.</exception>
```
Comment (antonfirsov) [inline xml/System.Net.Sockets/Socket.xml:4081]: Looks like `NotSupportedException` is missing here.

File: xml/System.Net.Sockets/Socket.xml [hunk 3]
```diff
@@ -4066,6 +4116,18 @@ This method populates the <xref:System.Net.Sockets.Socket> instance with data ga
...
```
Comment (antonfirsov) [inline xml/System.Net.Sockets/Socket.xml:4130]: Missing `NotSupportedException`.

File: xml/System.Net.Sockets/Socket.xml [hunk 4]
```diff
@@ -4103,6 +4165,17 @@ This method populates the <xref:System.Net.Sockets.Socket> instance with data ga
...
```
Comment (antonfirsov) [inline xml/System.Net.Sockets/Socket.xml:4178]: Missing `NotSupportedException`.

File: xml/System.Net.Sockets/Socket.xml [hunk 5]
```diff
@@ -4181,6 +4265,18 @@ This method populates the <xref:System.Net.Sockets.Socket> instance with data ga
...
```
Comment (antonfirsov) [inline xml/System.Net.Sockets/Socket.xml:4279]: Missing `NotSupportedException`.

File: xml/System.Net.Sockets/Socket.xml [hunk 6]
```diff
@@ -4313,6 +4409,17 @@ This method populates the <xref:System.Net.Sockets.Socket> instance with data ga
...
```
Comment (antonfirsov) [inline xml/System.Net.Sockets/Socket.xml:4422]: Missing `NotSupportedException`.

**Review by antonfirsov — COMMENTED** [review]
We should be consistent with `NotSupportedException`, otherwise looks good.

We should also review & fix the sync and the `Begin*` methods, but I will do that together with #5045, once this PR is merged.

**Review by liveans — APPROVED** [review]
LGTM

**Review by antonfirsov — APPROVED** [review]
LGTM

**Pull request merged**